<h1>Análise de Vendas de Cantina Escolar com Python: Um Estudo de Padrões de Consumo e Tendências</h1>

<h3>Situação problema:
A Cantina da Suzana, enfrenta desafios significativos na gestão de seu estoque devido à falta de clareza sobre quais produtos têm alta demanda e quais não estão vendendo bem. Essa situação impacta negativamente principalmente a vida social e econômica das pessoas envolvidas:

Impacto Social:

Desperdício de Alimentos: A falta de otimização no estoque leva ao desperdício de alimentos, um problema que afeta a comunidade ao desperdiçar recursos valiosos que poderiam ser melhor utilizados.
Frustração dos Alunos: A ausência de produtos populares pode gerar insatisfação entre os alunos, afetando negativamente sua experiência escolar.
Impacto Econômico:

Desperdício Financeiro: O desperdício de alimentos resulta em perdas financeiras significativas para a cantina, impactando a viabilidade econômica do empreendimento.
Ineficiência no Uso de Recursos: A falta de planejamento eficiente resulta em despesas desnecessárias e na perda de receitas potenciais, prejudicando a sustentabilidade financeira da cantina.</h3>

In [79]:
#importando a base de dados com pandas e exibindo a tabela

import pandas as pd
tabela = pd.read_excel("dados_venda.xlsx")
display(tabela)

data  dia da semana      mês          produto tipo de produto  \
0    2024-01-15  segunda-feria  janeiro          coxinha         salgado   
1    2024-01-15  segunda-feria  janeiro           empada         salgado   
2    2024-01-15  segunda-feria  janeiro           pastel         salgado   
3    2024-01-15  segunda-feria  janeiro        americano         salgado   
4    2024-01-15  segunda-feria  janeiro  cachorro-quente         salgado   
...         ...            ...      ...              ...             ...   
1273 2024-06-28    sexta-feira    junho           tortas            doce   
1274 2024-06-28    sexta-feira    junho            sucos          bebida   
1275 2024-06-28    sexta-feira    junho     refrigerante          bebida   
1276        NaT            NaN      NaN              NaN             NaN   
1277        NaT            NaN      NaN              NaN             NaN   

      valor unitário ($)  quantidade vendida  total vendido  
0                    5.0                15.0           75.0  
1                    4.5                22.0           99.0  
2                    6.0                20.0          120.0  
3                    7.0                30.0          210.0  
4                    8.0                25.0          200.0  
...                  ...                 ...            ...  
1273                 5.5                24.0          132.0  
1274                 4.0                36.0          144.0  
1275                 6.0                48.0          288.0  
1276                 NaN                 NaN            NaN  
1277                 NaN                 NaN       187723.0  

[1278 rows x 8 columns]

In [80]:
#visualizando as informações da tabela
display(tabela.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1278 entries, 0 to 1277
Data columns (total 8 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   data                1276 non-null   datetime64[ns]
 1   dia da semana       1276 non-null   object        
 2   mês                 1276 non-null   object        
 3   produto             1276 non-null   object        
 4   tipo de produto     1276 non-null   object        
 5   valor unitário ($)  1276 non-null   float64       
 6   quantidade vendida  1276 non-null   float64       
 7   total vendido       1277 non-null   float64       
dtypes: datetime64[ns](1), float64(3), object(4)
memory usage: 80.0+ KB


None

In [81]:
#visualizando as informações da tabela
display(tabela.describe().round(1))

valor unitário ($)  quantidade vendida  total vendido
count              1276.0              1276.0         1277.0
mean                  6.3                23.8          294.0
std                   1.7                11.9         5249.7
min                   4.0                 2.0           20.0
25%                   5.0                12.0           82.5
50%                   6.0                24.0          136.0
75%                   8.0                33.0          210.0
max                  10.0                56.0       187723.0

In [82]:
#tratando os dados da tabela importada
tabela = tabela.dropna()
display(tabela)

data  dia da semana      mês          produto tipo de produto  \
0    2024-01-15  segunda-feria  janeiro          coxinha         salgado   
1    2024-01-15  segunda-feria  janeiro           empada         salgado   
2    2024-01-15  segunda-feria  janeiro           pastel         salgado   
3    2024-01-15  segunda-feria  janeiro        americano         salgado   
4    2024-01-15  segunda-feria  janeiro  cachorro-quente         salgado   
...         ...            ...      ...              ...             ...   
1271 2024-06-28    sexta-feira    junho           salada  lanche natural   
1272 2024-06-28    sexta-feira    junho             bolo            doce   
1273 2024-06-28    sexta-feira    junho           tortas            doce   
1274 2024-06-28    sexta-feira    junho            sucos          bebida   
1275 2024-06-28    sexta-feira    junho     refrigerante          bebida   

      valor unitário ($)  quantidade vendida  total vendido  
0                    5.0                15.0           75.0  
1                    4.5                22.0           99.0  
2                    6.0                20.0          120.0  
3                    7.0                30.0          210.0  
4                    8.0                25.0          200.0  
...                  ...                 ...            ...  
1271                10.0                 5.0           50.0  
1272                 5.0                25.0          125.0  
1273                 5.5                24.0          132.0  
1274                 4.0                36.0          144.0  
1275                 6.0                48.0          288.0  

[1276 rows x 8 columns]

In [83]:
#avaliando o faturamento total dos 6 meses
faturamento_total = tabela["total vendido"]. sum()
print(faturamento_total)

187723.0


In [84]:
#avaliando o faturamento por mês
faturamento_por_mes = tabela[["mês", "total vendido"]]. groupby("mês"). sum()
display(faturamento_por_mes)

total vendido
mês                     
abril            35055.5
fevereiro        29778.0
janeiro          19699.0
junho            34205.5
maio             35834.5
março            33150.5

In [85]:
#avaliando o faturamento de cada produto em cada dia da semana do mês

# Lista de meses de interesse
meses = ["janeiro", "fevereiro", "março", "abril", "maio", "junho"]

# Dicionário para armazenar o resultado de cada mês
faturamento_por_mes = {}

# Iterar sobre cada mês e calcular o faturamento por produto e dia da semana
for mes in meses:
    faturamento_por_mes[mes] = tabela[tabela["mês"] == mes].groupby(["produto", "dia da semana"])["total vendido"].sum()

# Exibir os resultados
for mes, faturamento in faturamento_por_mes.items():
    print(f"Faturamento de {mes.capitalize()}:")
    print(faturamento)
    print("\n")


Faturamento de Janeiro:
produto            dia da semana
americano          quarta-feira     609.0
                   quinta-feira     448.0
                   segunda-feria    616.0
                   sexta-feira      462.0
                   terça-feria      616.0
bolo               quarta-feira     140.0
                   quinta-feira     105.0
                   segunda-feria    110.0
                   sexta-feira      190.0
                   terça-feria      130.0
cachorro-quente    quarta-feira     664.0
                   quinta-feira     512.0
                   segunda-feria    696.0
                   sexta-feira      568.0
                   terça-feria      560.0
coxinha            quarta-feira     140.0
                   quinta-feira      70.0
                   segunda-feria    175.0
                   sexta-feira       80.0
                   terça-feria      155.0
empada             quarta-feira     319.5
                   quinta-feira     220.5
                   

In [86]:
#importando a biblioteca plotly para avaliação gráfica

import plotly.express as px

In [87]:
#criando grafico para a quantidade de venda dos produtos em cada dia da semana.

# Criando a lista de meses de interesse
meses = ["janeiro", "fevereiro", "março", "abril", "maio", "junho"]

# Criando um DataFrame para armazenar todos os resultados
df_resultado = pd.DataFrame()

# Iterar sobre cada mês e calcular o faturamento por produto e dia da semana
for mes in meses:
    df_mes = tabela[tabela["mês"] == mes].groupby(["produto", "dia da semana"])["total vendido"].sum().reset_index()
    df_mes["mês"] = mes
    df_resultado = pd.concat([df_resultado, df_mes])

# Agora que os dados estão preparados, podemos criar um gráfico de barras
fig = px.bar(
    df_resultado,
    x="dia da semana",
    y="total vendido",
    color="produto",
    facet_col="mês",  # Criar gráficos separados por mês
    title="Faturamento por Produto e Dia da Semana",
    labels={"total vendido": "Faturamento", "dia da semana": "Dia da Semana"},
)

# Exibir o gráfico
fig.show()


In [88]:
# Avaliando a tendência de consumo dos produtos ao longo do tempo.
# Criando a lista de meses na ordem correta
meses = ["janeiro", "fevereiro", "março", "abril", "maio", "junho"]

# Ordenando os meses numericamente para manter a ordem correta no gráfico
ordem_meses = {mes: i for i, mes in enumerate(meses)}

# Adicionando uma coluna "ordem" no DataFrame original
tabela["ordem"] = tabela["mês"].map(ordem_meses)

# Agrupando por mês, produto e ordem para calcular o faturamento total por mês para cada produto
df_faturamento = tabela.groupby(["mês", "ordem", "produto"])["total vendido"].sum().reset_index()

# Ordenando os dados pelo campo "ordem" para garantir que os meses estejam na sequência correta
df_faturamento = df_faturamento.sort_values("ordem")

# Criando o gráfico de linha
fig = px.line(
    df_faturamento,
    x="mês",
    y="total vendido",
    color="produto",
    title="Tendência de Faturamento por Produto ao Longo do Tempo",
    labels={"total vendido": "Faturamento", "mês": "Mês"},
    markers=True  # Adicionar marcadores nos pontos da linha
)

# Exibindo o gráfico
fig.show()

In [89]:
#Avaliando a proporção de vendas por tipo de produto
# Agrupar os dados por 'tipos de produtos' e calcular o total vendido para cada tipo
df_pizza = tabela.groupby("tipo de produto")["total vendido"].sum().reset_index()

# Criando o gráfico de pizza
fig = px.pie(
    df_pizza, 
    values="total vendido", 
    names="tipo de produto", 
    title="Proporção de Vendas por Tipo de Produto",
    labels={"tipo de produto": "Tipo de Produto", "total vendido": "Total Vendido"},
)

# Exibir o gráfico
fig.show()


In [95]:
##Avaliando a proporção de vendas por tipo de produto em cada dia da semana.

# Agrupar os dados por 'dia da semana' e 'tipos de produtos', somando o total vendido para cada tipo de produto em cada dia
df_pizza_dias = tabela.groupby(["dia da semana", "tipo de produto"])["total vendido"].sum().reset_index()

# Criando gráficos de pizza por dia da semana usando facetas
fig = px.pie(
    df_pizza_dias, 
    values="total vendido", 
    names="tipo de produto", 
    title="Proporção de Vendas por Tipo de Produto e Dia da Semana",
    labels={"tipo de produto": "Tipo de Produto", "total vendido": "Total Vendido", "dia da semana": "Dia da Semana"},
    facet_col="dia da semana"  # Cria um gráfico de pizza para cada dia da semana
)

# Exibir o gráfico
fig.show()



<h1> <strong><em>Conclusão da avaliação </strong><em> </h1>

<h4> * Avaliando o faturamento por mês, observa-se que o mês que menos vendeu foi janeiro, e o outros meses tiveram vendas de 29778 (fevereiro) a 35834,5(maio).</h4>
    
<h4> * O produto pastel teve uma diminuição de consumo de 15,33% no mês de maio (em relação ao mês de abril) e de 25,67% no mês de junho (em relação ao mês de junho.</h4>
    
<h4> * O maior faturamento está realacionado aos produtos classificados como salgados, e desses o que possui menor faturamento é a coxinha, sendo que o maior é o cachorro-quente.</h4>    
    
<h4> * O menor faturamento está relacionado aos produtos classificados como doces.</h4>    
    
<h4>  * Avaliando o consumo dos dias da semana, observa-se que há um aumento siginificativo de 147% no consumo dos produtos classifcados como doces, quando se compara segunda-feira e sexta-feira. E há uma diminuição significativa de 66% no consumo dos produtos classficados como lanche natural, quando se compara segunda-feira e sexta-feira.</h4> 

<h1> <strong><em>Estratégias sugeridas </strong><em> </h1>
    
<h4> *Estratégias para o mês de janeiro: Como janeiro foi o mês com menor faturamento, a dona da cantina pode considerar campanhas promocionais ou pacotes especiais para atrair mais clientes durante esse mês. Pode ser interessante também avaliar se há fatores externos, como férias escolares, que impactam as vendas, e ajustar a estratégia para esses períodos.</h4> 

<h4> *Atenção ao consumo de pastéis: Houve uma queda significativa no consumo de pastéis em maio e junho. É importante investigar as causas dessa queda, como possíveis mudanças na qualidade, no preço, ou até mesmo concorrência. Com base nisso, a dona da cantina pode pensar em estratégias para recuperar as vendas desse produto, como melhorias na receita, promoções ou combos que incluam o pastel.</h4> 

<h4> *Foco nos salgados mais vendidos: Já que os salgados, especialmente o cachorro-quente, são os produtos com maior faturamento, pode ser interessante ampliar a oferta desses produtos ou criar variações que mantenham o interesse dos clientes. Quanto à coxinha, que tem o menor faturamento dentro dos salgados, talvez seja hora de repensar a estratégia de vendas desse item, seja ajustando o preço, oferecendo promoções ou melhorando a receita.</h4> 

<h4> *Incentivar o consumo de doces: Embora os doces tenham o menor faturamento geral, há um aumento significativo no consumo deles às sextas-feiras. Isso pode indicar uma oportunidade para promoções específicas nesse dia da semana ou para lançar novos produtos doces que possam atrair mais clientes em outros dias.</h4> 

<h4> *Promoções de lanches naturais: A queda no consumo de lanches naturais de segunda a sexta-feira sugere que os clientes podem estar optando por outras opções ao longo da semana. Oferecer promoções ou destacar os benefícios desses produtos, como opções mais saudáveis, pode ajudar a reverter essa tendência e estimular o consumo durante toda a semana.</h4> 